agent/src/main/scala/za/co/absa/atum/agent/model/Measurement.scala

```diff
@@ -68,13 +70,32 @@ object Measurement {
           )
       }
     }
-  }
 
-  /**
-   * When the Atum Agent itself performs the measurements, using Spark, then in some cases some adjustments are
-   * needed - thus we are converting the results to strings always - but we need to keep the information about
-   * the actual type as well.
-   */
-  case class MeasurementByAtum(measure: Measure, resultValue: String, resultType: ResultValueType.ResultValueType)
-    extends Measurement
+    def forCustomMeasure[T](measureName: String, controlCol: String, resultValue: T): MeasurementProvided[T] = {
```

Comment:
I'm probably missing something but I think that this function is not needed at all - the only difference (apart from the definition, but that can be easily changed) between this one and `apply` is that internal `handleSpecificType` - I would perhaps try to refactor this one instead. Maybe drop the whole `handleSpeficType` function altogether and/or move it elsewhere, check how I did it here: https://github.com/AbsaOSS/atum-service/pull/104/files#diff-c65609cc426a47499c22c7fde4375cb3e305ed9e0e601c8c6b8dfd2c72b38899

Reply:
But those two functions are almost the same let's not violate DRY principle

Reply:
Well, the handleSpecificType method on MeasurementProvided object expects to have resultValueType which is not really defined for CustomMeasure. If we change the CustomMeasure so that the client code will define it then we could use the mentioned method. But I don't quite see why would we do that cause in my opinion that would just complicate the creation of CustomMeasure. Moreover the client code cares about creation of MeasurementProvided, not the custom measure itself. For that reason I construct the MeasurementProvided in the forCustomMeasure method and use the provided result value's to determine resultValueType (for the MeasurementProvided). From the client code perspective I feel that's just a simplest way how to create a MeasurementProvided. 

```scala
MeasurementProvided.forCustomMeasure("customMeasureName", "col", BigDecimal(1))

Reply:
I agree but I still think there is a way. But why not to at least change

`def forCustomMeasure[T](measureName: String, controlCol: String, resultValue: T): MeasurementProvided[T]`

to

`def apply[T](measure: CustomMeasure, resultValue: T): MeasurementProvided[T]`

❓ 

It would be closer to the `def apply[T](measure: AtumMeasure, resultValue: T): Measurement` 'API'

Reply:
Renamed the forCustomMeasure method to apply.